util.py

```diff
@@ -0,0 +1,12 @@
+def convert_str_to_bool(value: str, default: bool) -> bool:
+    try:
+        if value is None:
+            return default
+        if value.lower() == 'true':
+            return True
+        elif value.lower() == 'false':
+            return False
+        else:
+            return default
+    except Exception as e:
+        return default
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2413706579>#b6ea43</a><span> - 10/15/2024, 05:29 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function catches all exceptions silently and returns the default value without logging or providing any information about the error. This may hide unexpected issues and make debugging difficult.
</div>
<div id="fix">
<b>Fix:</b> Instead of catching all exceptions, catch only the specific exceptions you expect (e.g., AttributeError for the .lower() method). For unexpected exceptions, log them before returning the default value to maintain the function's contract while still providing visibility into unexpected issues.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -11,2 +11,6 @@
 -    except Exception as e:
 -        return default
 +    except AttributeError:
 +        return default
 +    except Exception as e:
 +        import logging
 +        logging.error(f"Unexpected error in convert_str_to_bool: {e}")
 +        return default
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged